tools/ci/gha-install-njoy.sh

```diff
@@ -1,7 +1,13 @@
 #!/bin/bash
 set -ex
+
+# NJOY variables
+NJOY_REPO='https://github.com/njoy/NJOY2016'
+NJOY_INSTALL_DIR=$HOME/NJOY2016
```

Comment:
It might be nice to put all the installed dependencies in a single folder, e.g., `$HOME/dependencies/NJOY2016`. Also, in the case of NJOY (and probably the others), we should just be caching the installed version. Right now, the cache for NJOY is 1 GB when it should be much less.

Reply:
I've trimmed the NJOY repository by ~30% by removing its own NJOY-specific testing suite out of the cache; however, removing the build directory causes issues with the vectfit installation. Aside from NJOY and vectfit, all other of the dependencies at the very least remove their own build directories before caching.

I would like to move all the installed dependencies into a single folder as well; however, I will address that in a future PR since it would take a lot of testing and plenty of commits (and this PR already has enough of those as is!).